---
title: "Detection_plot"
author: "Carrie Wright"
date: "June 6, 2018"
output: html_document
---
---
title: "ComparsionofAmt"
author: "Carrie Wright"
date: "2/13/2018"
output: html_document
---


###Prepare for UMI tools Shell Script on SRV4: /media/Backup1_/smallRNA/FullHiSeq_mismatch0/accuracy/samples_lanesCombined/trimmed_fastq/UMI_tools_bespoke_code.sh
```{r, engine='bash', eval = FALSE, echo=FALSE}
mkdir UMI_duplicates_rem

InDir=/media/Backup1_/smallRNA/FullHiSeq_mismatch0/repro/samples_lanesCombined/trimmed_fastq
outDir=/media/Backup1_/smallRNA/FullHiSeq_mismatch0/repro/samples_lanesCombined/trimmed_fastq/UMI_duplicates_rem

#reading every 4th line starting with line 2, get first 4 characters of sequence
#awk2='NR%4==2'
#< list_for_UMI.txt parallel -P4 "cat $InDir/mm0_acc_NEXT_trim1.{}_R1.fq | awk '$awk2' | cut -d' ' -f2 | cut -c1-4 > $outDir/first4_{}.txt"

#reading every 4th line starting with line 2, get last 4 characters of sequence
#< list_for_UMI.txt parallel -P4 "cat $InDir/mm0_acc_NEXT_trim1.{}_R1.fq | awk '$awk2' | sed 's/^.*\(.\{4\}\)/\1/' > $outDir/last4_{}.txt"

#pasting first UMI 4 nuc. with last UMI 4 nuc.
#< list_for_UMI.txt parallel -P4 "paste -d'\0' $outDir/first4_{}.txt $outDir/last4_{}.txt > $outDir/UMI_{}.txt"

#quadruple UMIs
#< list_for_UMI.txt parallel -P4 "awk '{for(i=0;i<4;i++)print}' $outDir/UMI_{}.txt >$outDir/quad_UMI_{}.txt"

# add an "_" to the front of every UMI line
#awk3='$0="_"$0'
#< list_for_UMI.txt parallel -P4 "awk '$awk3'  $outDir/quad_UMI_{}.txt > $outDir/final_UMI_{}.txt"

# add the UMI to the fastq file identifier line
#awk4='{getline p<f} (NR%4==1){$1=$1" "$2;$2=p}1'
#< list_for_UMI.txt parallel -P4 "awk '$awk4' OFS= f=$outDir/final_UMI_{}.txt $InDir/mm0_acc_NEXT_trim1.{}_R1.fq > $outDir/NEXT_{}_UMItools_R1.fq"

#remove reads from fastq with Ns in the UMI: DID NOT RUN THIS COMMAND!!!!!!!!!!!!!!!!!!!!
#< list_for_UMI.txt parallel -P4 "sed -e '/_N\|_.*N/,+3d' $outDir/NEXT_{}_UMItools_R1.fq > $outDir/NEXT_Ns_rem_{}_UMItools_R1.fq"

#remove random 4 base pair seqs that make up the UMI from the fastq read sequence line:
< list_for_UMI.txt parallel -P4 "cutadapt -u 4 -o $outDir/trim2_{}_Ns_kept_forUMI_tools.fq $outDir/NEXT_{}_UMItools_R1.fq"

< list_for_UMI.txt parallel -P4 "cutadapt -m 18 -u  -4 -o $outDir/trimmed_{}_Ns_kept_forUMI_tools.fq $outDir/trim2_{}_Ns_kept_forUMI_tools.fq"


#remove space form the identifier of the fastq
< list_for_UMI.txt parallel -P4 "sed 's/ /-/' $outDir/trimmed_{}_Ns_kept_forUMI_tools.fq > $outDir/nospace_trimmed_{}_Ns_kept_forUMI_tools.fq"

#bowtie allignment
< list_for_UMI.txt parallel -P3 "/usr/bin/bowtie /media/DATA/carrie/miRge/miRge-master/miRge.seqLibs/human/mirna --fullref -n 0 -S $outDir/nospace_trimmed_{}_Ns_kept_forUMI_tools.fq $outDir/NEXT_{}_Ns_kept_readyforUMItools.sam"

#convert to bams
< list_for_UMI.txt parallel -P3 "samtools view -bS -o $outDir/NEXT_{}_Ns_kept_readyforUMItools.bam $outDir/NEXT_{}_Ns_kept_readyforUMItools.sam"

#index and sort bams
< list_for_UMI.txt parallel -P3 "samtools sort $outDir/NEXT_{}_Ns_kept_readyforUMItools.bam $outDir/NEXT_{}_Ns_kept_readyforUMItools_sorted"
< list_for_UMI.txt parallel -P3 "samtools index $outDir/NEXT_{}_Ns_kept_readyforUMItools_sorted.bam"

#UMItools
< list_for_UMI.txt parallel -P3 "umi_tools dedup --method directional -I $outDir/NEXT_{}_Ns_kept_readyforUMItools_sorted.bam -S $outDir/directional_deduped_Ns_kept_{}_UMItools.bam"


#convert deduped bam files to fastq files
<list_for_UMI.txt parallel -P3 "bam2fastx -q -Q -A -o $outDir/directional_dedupped_Ns_kept_{}_bam2fastq.fq $outDir/directional_deduped_Ns_kept_{}_UMItools.bam"


#same but adjacency method
#< list_for_UMI.txt parallel -P3 "umi_tools dedup --method adjacency -I /media/Backup1_/smallRNA/FullHiSeq_mismatch0/accuracy/samples_lanesCombined/trimmed_fastq/UMI_duplicates_rem/NEXT_acc_{}_readyforUMItools_sorted.bam -S /media/Backup1_/smallRN
A/FullHiSeq_mismatch0/accuracy/samples_lanesCombined/trimmed_fastq/UMI_duplicates_rem/adjacency_deduped_acc_{}_UMItools.bam"
#<list_for_UMI.txt parallel -P3 "bam2fastx -q -Q -A -o $outDir/adjacency_dedupped_acc_{}_bam2fastq.fq $outDir/adjacency_deduped_acc_{}_UMItools.bam"

#same but unique method
#< list_for_UMI.txt parallel -P3 "umi_tools dedup  --method unique -I /media/Backup1_/smallRNA/FullHiSeq_mismatch0/accuracy/samples_lanesCombined/trimmed_fastq/UMI_duplicates_rem/NEXT_acc_{}_readyforUMItools_sorted.bam -S /media/Backup1_/smallRNA/
FullHiSeq_mismatch0/accuracy/samples_lanesCombined/trimmed_fastq/UMI_duplicates_rem/unique_deduped_acc_{}_UMItools.bam"
#<list_for_UMI.txt parallel -P3 "bam2fastx -q -Q -A -o $outDir/unique_dedupped_acc_{}_bam2fastq.fq $outDir/unique_deduped_acc_{}_UMItools.bam"

```

###miRge analysis on SRV2:/miRge/miRge-master: #first to run all 1000ng samples...then going to run all samples, except synthetic
```{r,engine='bash', eval = FALSE, echo=FALSE}

perl miRge.pl --species human --diff-isomirs --phred64 --bowtie /usr/bin/bowtie --CPU 10 --SampleFiles mm0_acc_Clontech_acc_trimmed.1_R1.fq,mm0_acc_Clontech_acc_trimmed.2_R1.fq,mm0_acc_Clontech_acc_trimmed.3_R1.fq,mm0_acc_Illumina_trimmed.1_R1.fq,mm0_acc_Illumina_trimmed.2_R1.fq,mm0_acc_Illumina_trimmed.3_R1.fq,mm0_acc_NEB_trimmed.1_R1.fq,mm0_acc_NEB_trimmed.2_R1.fq,mm0_acc_NEB_trimmed.3_R1.fq,mm0_acc_NEXT_trimmed.1_R1.fq,mm0_acc_NEXT_trimmed.2_R1.fq,mm0_acc_NEXT_trimmed.3_R1.fq,directional_dedupped_acc_Ns_kept_1_bam2fastq.fq,directional_dedupped_acc_Ns_kept_2_bam2fastq.fq,directional_dedupped_acc_Ns_kept_3_bam2fastq.fq,mm0_Clontech_trimmed.1_R1.fq,mm0_Clontech_trimmed.2_R1.fq,mm0_Clontech_trimmed.3_R1.fq,mm0_Clontech_trimmed.4_R1.fq,mm0_Clontech_trimmed.5_R1.fq,mm0_Clontech_trimmed.6_R1.fq,mm0_Clontech_trimmed.7_R1.fq,mm0_Clontech_trimmed.8_R1.fq,mm0_Clontech_trimmed.9_R1.fq,mm0_Clontech_trimmed.10_R1.fq,mm0_Clontech_trimmed.11_R1.fq,mm0_Clontech_trimmed.12_R1.fq,mm0_Clontech_trimmed.13_R1.fq,mm0_Clontech_trimmed.14_R1.fq,mm0_Clontech_trimmed.15_R1.fq,mm0_Clontech_trimmed.16_R1.fq,mm0_Clontech_trimmed.17_R1.fq,mm0_Clontech_trimmed.18_R1.fq,mm0_Illumina_trimmed.1_R1.fq,mm0_Illumina_trimmed.2_R1.fq,mm0_Illumina_trimmed.3_R1.fq,mm0_Illumina_trimmed.4_R1.fq,mm0_Illumina_trimmed.5_R1.fq,mm0_Illumina_trimmed.6_R1.fq,mm0_Illumina_trimmed.7_R1.fq,mm0_Illumina_trimmed.8_R1.fq,mm0_Illumina_trimmed.9_R1.fq,mm0_NEB_trimmed.1_R1.fq,mm0_NEB_trimmed.2_R1.fq,mm0_NEB_trimmed.3_R1.fq,mm0_NEB_trimmed.4_R1.fq,mm0_NEB_trimmed.5_R1.fq,mm0_NEB_trimmed.6_R1.fq,mm0_NEB_trimmed.7_R1.fq,mm0_NEB_trimmed.8_R1.fq,mm0_NEB_trimmed.9_R1.fq,mm0_NEB_trimmed.10_R1.fq,mm0_NEB_trimmed.11_R1.fq,mm0_NEB_trimmed.12_R1.fq,mm0_NEXT_trimmed.1_R1.fq,mm0_NEXT_trimmed.2_R1.fq,mm0_NEXT_trimmed.3_R1.fq,mm0_NEXT_trimmed.4_R1.fq,mm0_NEXT_trimmed.5_R1.fq,mm0_NEXT_trimmed.6_R1.fq,mm0_NEXT_trimmed.7_R1.fq,mm0_NEXT_trimmed.8_R1.fq,mm0_NEXT_trimmed.9_R1.fq,mm0_NEXT_trimmed.10_R1.fq,mm0_NEXT_trimmed.11_R1.fq,mm0_NEXT_trimmed.12_R1.fq,mm0_NEXT_trimmed.13_R1.fq,mm0_NEXT_trimmed.14_R1.fq,mm0_NEXT_trimmed.15_R1.fq,mm0_NEXT_trimmed.16_R1.fq,mm0_NEXT_trimmed.17_R1.fq,mm0_NEXT_trimmed.18_R1.fq,directional_dedupped_Ns_kept_1_bam2fastq.fq,directional_dedupped_Ns_kept_2_bam2fastq.fq,directional_dedupped_Ns_kept_3_bam2fastq.fq,directional_dedupped_Ns_kept_4_bam2fastq.fq,directional_dedupped_Ns_kept_5_bam2fastq.fq,directional_dedupped_Ns_kept_6_bam2fastq.fq,directional_dedupped_Ns_kept_7_bam2fastq.fq,directional_dedupped_Ns_kept_8_bam2fastq.fq,directional_dedupped_Ns_kept_9_bam2fastq.fq,directional_dedupped_Ns_kept_10_bam2fastq.fq,directional_dedupped_Ns_kept_11_bam2fastq.fq,directional_dedupped_Ns_kept_12_bam2fastq.fq,directional_dedupped_Ns_kept_13_bam2fastq.fq,directional_dedupped_Ns_kept_14_bam2fastq.fq,directional_dedupped_Ns_kept_15_bam2fastq.fq,directional_dedupped_Ns_kept_16_bam2fastq.fq,directional_dedupped_Ns_kept_17_bam2fastq.fq,directional_dedupped_Ns_kept_18_bam2fastq.fq

```

###load data into R
```{r, echo=FALSE}
library(here)

Pheno<- read.table(here("Brain_samples_missingClontech_batch2/Pheno_2.csv"), header = T, sep = ",")
miR_counts <- read.table(here("Brain_samples_missingClontech_batch2/miR.Counts.csv"),header = T, sep = ",")



miRNAtotal <-t(miR_counts[1,])
miRNAtotal <- miRNAtotal[-1,]
miRNAtotal<-data.frame(miRNAtotal)
miR_counts <-miR_counts[-1,]
miR_counts <-miR_counts[-1]

#######################

colnames(miR_counts)<-gsub("directional_dedupped|directional_deduped", "Deduped", colnames(miR_counts))
colnames(miR_counts)<-gsub("NEXT_", "NEXTflex_", colnames(miR_counts))

miR_counts<-miR_counts[,-grep("Accur", Pheno$Batch)]
Pheno<-Pheno[-grep("Accur", Pheno$Batch),]


library(XML)
library(RCurl)

# report_url<-here("report.html")
# report<-readHTMLTable(doc = report_url, header = TRUE) #need to read every third line
# report <-as.data.frame(report)
# report <-report[1:9]
# 
# report <-report[seq(from =1, to = nrow(report), by = 5),]
# total_reads <-as.numeric(as.character(report$NULL.Total.Input.Reads))
# report$miRNAtotalreads <- miRNAtotal
# report$percentmiRNA <-as.vector((miRNAtotal/total_reads)*100)
# report$percentnoncoding <-as.vector((as.numeric(as.character(report$NULL.Other.Noncoding.RNA.Reads))/total_reads)*100)
# report$percentmRNA<-as.vector((as.numeric(as.character(report$NULL.mRNA.Reads))/total_reads)*100)
# report$percenthairpin <-as.vector((as.numeric(as.character(report$NULL.Hairpin.Reads))/total_reads)*100)
# percentages <-report[grep("percent", colnames(report))]
# rownames(percentages) <- report$NULL.File.name.s.
# percentages$total <- rowSums(percentages)
```

###DESeq2
```{r, eval=TRUE, warning=FALSE, message=FALSE, echo =FALSE}
library(DESeq2)
library(reshape2)
library(ggplot2)
norm_miR <-list()
dds<-DESeqDataSetFromMatrix(countData = miR_counts, colData = Pheno, design = ~ Kit)
dds <- estimateSizeFactors(dds)
norm_miR<-data.frame(counts(dds, normalized = TRUE))

#load(here("Complete_data/split_kit_1batch.rda"))
get_det <-function(x){sum(x>10)}#determine number of miRNAs with greater than 10 reads
detected <-lapply(norm_miR, get_det)
detected <-data.frame(unlist(detected))
det_Data <- cbind(detected, Pheno$startingAmt, Pheno$Kit)
det_Data$`Pheno$startingAmt`<- factor(det_Data$`Pheno$startingAmt`)
Det_data<-melt(det_Data)
Det_data$`Pheno$startingAmt` <-as.numeric(as.character(det_Data$`Pheno$startingAmt`))
colnames(Det_data)<- c("Starting_Amount", "Kit", "varaible", "miRNA")

Det_data$Kit <-factor(Det_data$Kit, levels  =c("Clontech","Illumina","NEB","NEXTflex","Deduped", "Fivepercent"))
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

n = 6
cols = gg_color_hue(n)
ggplot(data = Det_data, aes(x = Starting_Amount, y = miRNA, color = Kit)) + geom_point() +facet_grid(.~Kit) + scale_x_continuous(breaks=unique(Det_data$Starting_Amount)) + theme(legend.position = "none") + labs(y = " Number of Uniquely Detected miRNA Sequences", title = "miRNA Detection across kits") +geom_smooth(method = "lm", se=TRUE, color="black", aes(group=1))

ggplot(data = Det_data, aes(x = Starting_Amount, y = miRNA, group = Starting_Amount)) + geom_jitter(aes(color = Kit))+ geom_boxplot(aes(fill = Kit)) +scale_fill_manual(values=cols)+ facet_grid(.~Kit) + scale_x_continuous(breaks=unique(Det_data$Starting_Amount)) + theme(legend.position = "none", axis.text.x = element_text(angle = 90, hjust = 1)) + labs(y = " Number of Uniquely Detected miRNA Sequences", title = "miRNA Detection across kits") +geom_smooth(method = "lm", se=TRUE, color="black", aes(group=1, fill = Kit))

ggplot(data = Det_data, aes(x = Starting_Amount, y = miRNA, group = Starting_Amount)) + geom_jitter(aes(color = Kit))+ geom_boxplot(aes(fill = Kit)) +scale_fill_manual(values=cols)+ facet_grid(.~Kit)+theme(strip.text.x = element_text(size = 16)) + scale_x_continuous(breaks=unique(Det_data$Starting_Amount)) + theme(legend.position = "none", axis.text.x = element_text(size =14, angle = 60, hjust = 1), axis.text.y = element_text(size =14, hjust = 1), axis.title=element_text(size = 20), plot.title = element_text(size = 20)) + labs(y = " Number of Uniquely Detected miRNA Sequences", title = "miRNA Detection across kits") +geom_smooth(method = "loess", se=TRUE, color="black", aes(group=1, fill = Kit))

#to add formula   
#https://stackoverflow.com/questions/7549694/adding-regression-line-equation-and-r2-on-graph
```

#stats
Is there a relationship between starting amount and the number detected
```{r}
library(broom)
det_mod <-lm(Det_data$miRNA ~ Det_data$Starting_Amount)
summary(det_mod)
det_mod <-lm(Det_data$miRNA ~ Det_data$Kit)
summary(det_mod)

det_mod <-lm(Det_data$miRNA ~ Det_data$Kit*Det_data$Starting_Amount)
summary(det_mod)

split_kit <-list()
for(i in Det_data$Kit) { 
  split_kit[[i]] <- data.frame(Det_data[which(Det_data$Kit==i),])
} 
#Det_data$Starting_Amount<-factor(Det_data$Starting_Amount)
 run_model <-function(x){(lm(x[["miRNA"]]~x[["Starting_Amount"]]))}
model_stats<-lapply(split_kit,run_model)
model_summary<-lapply(model_stats, summary)
lapply(model_summary, glance)# only Deduped and Fivepercent are significant


split_amt <-list()
for(i in unique(as.character(Det_data$Starting_Amount))) { 
  split_amt[[i]] <- data.frame(Det_data[which(Det_data$Starting_Amount==i),])
} 
 run_model <-function(x){(lm(x[["miRNA"]]~x[["Kit"]]))}
model_stats<-lapply(split_amt,run_model)
model_summary<-lapply(model_stats, summary)
lapply(model_summary, glance)# all significant


t.test(split_kit$Clontech$miRNA, split_kit$NEXTflex$miRNA)

t.test(split_amt$`100`$miRNA[grep("Clontech",split_amt$`100`$Kit)], split_amt$`100`$miRNA[grep("NEB", split_amt$`100`$Kit)])

Det <-t(detected)
colnames(Det)<- Pheno$Kit
#split_amt <-list()
#for(i in unique(Pheno$startingAmt)) { 
#    split_amt[[i]] <- data.frame(Det[,which(Pheno$startingAmt==i), drop = FALSE])
#    colnames(split_amt[[i]]) <- Pheno$Kit[which(Pheno$startingAmt==i)]}


```


```{r, echo = TRUE, eval=TRUE}
library(dplyr)
get_test_names <- function(data){
  test_names <<- data.frame(combn(unique(names(data)), m= 2))
}

get_test_results<- function(data,test_names) {
  tresults<<-list()
  tested_names1<<-list()
  tested_names2<<-list()
  for(i in names(test_names)){
    #tested_names[[i]]<<-(test_names[i][,1])
    Kit1<-data[grep(test_names[i][1,], names(data))]
    Kit2<-data[grep(test_names[i][2,], names(data))]
    #Kit1<-data.frame(select(data, names(data)[names(data) %in% test_names[i][1,]]))
    #Kit2<-data.frame(select(data, names(data)[names(data) %in% test_names[i][2,]]))
    tested_names1[[i]]<<-names(Kit1)
    tested_names2[[i]]<<-names(Kit2)
    # colnames(Kit1)<-c("error")
    # colnames(Kit2)<-c("error")
    tresults[[i]]<<-t.test(x=Kit1[[1]][4], y=Kit2[[1]][4], paired = FALSE) ### may have messed things up adding paired = TRUE previously had more ))
    tested_kits <<-paste0(tested_names1, "&", tested_names2)
  }
}

get_ttestStats<- function(x) {
  #print(length(test_names))
  c(t =format(x$statistic, digits = 2),
    df = format(x$parameter, digits = 0),
    p.value = format(x$p.value, scientific = TRUE, digits = 2),
    bonferroni_thresh = format(.05/length(test_names), digits = 2),
    sig = ifelse(x$p.value<(.05/length(test_names)), "yes", "no"))
}

#get_lmStats <-function(x) {
#  c(p=x$coefficients[2,4], scientific = TRUE, digits =2)
#}
```

```{r}
get_test_names(split_kit)
get_test_results(data = split_kit, test_names = test_names)
ttestStats_across<-data.frame(lapply(tresults, get_ttestStats))
colnames(ttestStats_across)<-tested_kits
ttestStats_across
```
#detected miRNAs
```{r, echo = FALSE, eval = FALSE}
get_miRNAs <- function(x) {rownames(as.data.frame(x))}
miRNA_det_100<-lapply(split_amt$`100`,get_miRNAs)
lengths<-lapply(miRNA_det_100, length)
lengths<- paste0(lengths, "\n")
names(miRNA_det_100) <-paste0(names(miRNA_det_100), "\n")
names(miRNA_det_100) <-paste0(names(miRNA_det_100), lengths)

miRNA_det_250<-lapply(split_amt$`250`,get_miRNAs)
lengths<-lapply(miRNA_det_250, length)
names(miRNA_det_250) <-paste0(names(miRNA_det_250), "\n")
names(miRNA_det_250) <-paste0(names(miRNA_det_250), lengths)

miRNA_det_500<-lapply(split_amt$`500`,get_miRNAs)
lengths<-lapply(miRNA_det_500, length)
names(miRNA_det_500) <-paste0(names(miRNA_det_500), "\n")
names(miRNA_det_500) <-paste0(names(miRNA_det_500), lengths)

miRNA_det_1000<-lapply(split_amt$`1000`,get_miRNAs)
lengths<-lapply(miRNA_det_1000, length)
names(miRNA_det_1000) <-paste0(names(miRNA_det_1000), "\n")
names(miRNA_det_1000) <-paste0(names(miRNA_det_1000), lengths)

miRNA_det_1500<-lapply(split_amt$`1500`,get_miRNAs)
lengths<-lapply(miRNA_det_1500, length)
names(miRNA_det_1500) <-paste0(names(miRNA_det_1500), "\n")
names(miRNA_det_1500) <-paste0(names(miRNA_det_1500), lengths)

miRNA_det_2000<-lapply(split_amt$`2000`,get_miRNAs)
lengths<-lapply(miRNA_det_2000, length)
names(miRNA_det_2000) <-paste0(names(miRNA_det_2000), "\n")
names(miRNA_det_2000) <-paste0(names(miRNA_det_2000), lengths)
```
VennDiagram
```{r, echo = FALSE, eval = FALSE}
library(VennDiagram)
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}

n = 6
cols = gg_color_hue(n)
cols <-c(cols[1], cols[3:6])

Amount <-data.frame(paste0("amt_", det_Data$`Pheno$startingAmt`))
vp_100 <- venn.diagram(det_Data$unlist.detected.[grep("100", Amount$paste0..amt_...det_Data..Pheno.startingAmt..)], fill = c(cols), alpha = 0.5, filename = NULL, margin = 0.2, cat.dist = c(.25,.25,.15, .15, .25))
vp_250 <- venn.diagram(miRNA_det_250, fill = c(cols), alpha = 0.5, filename = NULL, margin = 0.2, cat.dist = c(.25,.25,.15, .15, .25))
vp_500 <- venn.diagram(miRNA_det_500, fill = c(cols), alpha = 0.5, filename = NULL, margin = 0.2, cat.dist = c(.25,.25,.15, .15, .25))
n=5
cols = gg_color_hue(n)
vp_1000 <- venn.diagram(miRNA_det_1000[1:5], fill = c(cols), alpha = 0.5, filename = NULL, margin = 0.2, cat.dist = c(.25,.25,.25, .25, .25))

cols <-c(cols[1:2], cols[4:6])
vp_1500 <- venn.diagram(miRNA_det_1500, fill = c(cols), alpha = 0.5, filename = NULL, margin = 0.2, cat.dist = c(.25,.25,.25, .25, .25))
vp_2000 <- venn.diagram(miRNA_det_2000, fill = c(cols), alpha = 0.5, filename = NULL, margin = 0.2, cat.dist = c(.25,.25,.25, .25, .25))
```

need to run seperately 
grid.draw(vp_100);
grid.draw(vp_250);
grid.draw(vp_500);

grid.draw(vp_1000)
grid.draw(vp_1500)
grid.draw(vp_2000)